### PR TITLE
Added few tests and required docstring

### LINKED ISSUE #4231: add a test for complex integral from wikipedia
bc.. In [4]: d = (1+2_x+sqrt(x+log(x))_(1+3_x) +
x__2)/(x_(x+sqrt(x+log(x)))*sqrt(x+log(x)))

In [5]: d
Out[5]:
            ⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽              2
1 + 2⋅x + ╲╱ x + log(x) ⋅(1 + 3⋅x) + x
───────────────────────────────────────
   ⎛      ⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎞   ⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽
 x⋅⎝x + ╲╱ x + log(x) ⎠⋅╲╱ x + log(x)

In [6]: integrate(d, x)
Out[6]:
     ⎛      ⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎞       ⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽⎽
2⋅log⎝x + ╲╱ x + log(x) ⎠ + 2⋅╲╱ x + log(x)

more info here: http://groups.google.com/group/sympy/browse_thread/thread/9a61d681e6f967e8

p. Original issue for "#4231":https://github.com/sympy/sympy/issues/4231: "http://code.google.com/p/sympy/issues/detail?id=1132":http://code.google.com/p/sympy/issues/detail?id=1132

p. Original author: "https://code.google.com/u/104039945248245758823/":https://code.google.com/u/104039945248245758823/

### LINKED ISSUE #5101: has() doesn't like sqrt
While it's possible to see if an expr has cos, a similar request for sqrt fails:
```
>>> cos(x).has(cos)
True
>>> sqrt(x).has(sqrt)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\core\basic.py", line 916, in has
    p = sympify(patterns[0])
  File "sympy\core\sympify.py", line 125, in sympify
    return ast_parser.parse_expr(a, locals)
  File "sympy\core\ast_parser.py", line 90, in parse_expr
    raise SympifyError("Cannot parse.")
sympy.core.sympify.SympifyError: SympifyError: 'Cannot parse.'
```

Original issue for #5101: http://code.google.com/p/sympy/issues/detail?id=2002
Original author: https://code.google.com/u/117933771799683895267/
Original owner: https://code.google.com/u/117933771799683895267/

### LINKED ISSUE #9824: solveset fails on sin(x)**2 - 2*sin(x) + 1
`solveset` raises `NotImplementedError` on the above equation.

However, a similar variation is solved :
`>>> solveset(cos(x)**2 - 2*cos(x) + 1, x, domain=S.Reals)`
`ImageSet(Lambda(_n, 2*_n*pi), Integers())`

### PR DESCRIPTION
**Pull request by Smit-create**
Added few tests and required docstring
#### References to other Issues or PRs

Fixes #5101
Fixes #4231
Fixes #9824 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/functions/elementary/miscellaneous.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def sqrt(arg, evaluate=None):
     >>> [rootof(x**2-3,i) for i in (0,1)]
     [-sqrt(3), sqrt(3)]
 
+    sqrt(x) is Pow(x, S.Half), therefore `sqrt(x).has(sqrt)` doesn't work with
```
Comment (sylee957) [inline sympy/functions/elementary/miscellaneous.py]: You should not use single backticks, but double backticks for highlighting.
But I think that this should better be documented with a runnable code sample about why it is not working.

**Comment by smichr** [comment]
Maybe a gotchas entry should be added:
```diff
diff --git a/doc/src/gotchas.rst b/doc/src/gotchas.rst
index b7e2a7d..977f9b5 100644
--- a/doc/src/gotchas.rst
+++ b/doc/src/gotchas.rst
@@ -568,6 +568,28 @@ of :obj:`~.asin`, :obj:`~.acos` and so on instead of the usual ``arcsin``
 and ``arccos``.  Use the methods described in :ref:`Symbols <symbols>`
 above to see the names of all SymPy functions.

+Sqrt is not a Function
+----------------------
+
+There is no ``sqrt`` function in the same way that there is an
+exponential function (``exp``). ``sqrt(x)`` is used to represent
+``Pow(x, S(1)/2)`` so if you want to know if an expression has any
+square roots in it, ``expr.has(sqrt)`` will not work. You must look
+for ``Pow`` with an exponent of one half (or negative one half if it
+is in a denominator, e.g.
+
+    >>> (y + sqrt(x)).find(Wild('w')**S.Half)
+    {sqrt(x)}
+    >>> (y + 1/sqrt(x)).find(Wild('w')**-S.Half)
+    {1/sqrt(x)}
+
+If you are interested in any power of the ``sqrt`` then the
+following pattern would be appropriate
+
+    >>> sq = lambda s: s.is_Pow and s.exp.is_Rational and s.exp.q == 2
+    >>> (y + sqrt(x)**3).find(sq)
+    {x**(3/2)}
+
 Special Symbols
 ===============
```

**Comment by Smit-create** [comment]
Yes I will add them. Thanks!

File: sympy/functions/elementary/miscellaneous.py [hunk 2]
```diff
@@ -115,6 +115,17 @@ def sqrt(arg, evaluate=None):
     >>> [rootof(x**2-3,i) for i in (0,1)]
     [-sqrt(3), sqrt(3)]
 
+    sqrt(x) is Pow(x, S.Half), therefore ``sqrt(x).has(sqrt)`` doesn't work
```
Comment (smichr) [inline sympy/functions/elementary/miscellaneous.py]: or
```
Although ``sqrt`` is printed, there is no ``sqrt`` function so looking for
``sqrt`` in an expression will fail:

    >>> sqrt(x).has(sqrt)
    Traceback (most recent call last):
      ...
    SyntaxError: invalid syntax

To find ``sqrt`` look for ``Pow`` with an exponent of ``1/2``:

    >>> (x + 1/sqrt(x)).find(lambda i: i.is_Pow and abs(i.exp) is S.Half)
    {1/sqrt(x)}
```

Comment (Smit-create) [reply sympy/functions/elementary/miscellaneous.py]: Surely, Would go for that.

**Comment by smichr** [comment]
Let's see if others have comments.

**Comment by Smit-create** [comment]
@smichr @sylee957  Thanks :)

**Pull request merged**